demos/multi_channel/yolo_v3/main.cpp

```diff
@@ -0,0 +1,584 @@
+        this->ymax = static_cast<int>(this->ymin + h * h_scale);
+        this->class_id = class_id;
+        this->confidence = confidence;
+    }
+
+    bool operator <(const DetectionObject &s2) const {
+        return this->confidence < s2.confidence;
+    }
+    bool operator >(const DetectionObject &s2) const {
+        return this->confidence > s2.confidence;
+    }
+};
+
+double IntersectionOverUnion(const DetectionObject &box_1, const DetectionObject &box_2) {
+    double width_of_overlap_area = fmin(box_1.xmax, box_2.xmax) - fmax(box_1.xmin, box_2.xmin);
+    double height_of_overlap_area = fmin(box_1.ymax, box_2.ymax) - fmax(box_1.ymin, box_2.ymin);
+    double area_of_overlap;
+    if (width_of_overlap_area < 0 || height_of_overlap_area < 0)
+        area_of_overlap = 0;
+    else
+        area_of_overlap = width_of_overlap_area * height_of_overlap_area;
+    double box_1_area = (box_1.ymax - box_1.ymin)  * (box_1.xmax - box_1.xmin);
+    double box_2_area = (box_2.ymax - box_2.ymin)  * (box_2.xmax - box_2.xmin);
+    double area_of_union = box_1_area + box_2_area - area_of_overlap;
+    return area_of_overlap / area_of_union;
+}
+
+void ParseYOLOV3Output(const InferenceEngine::CNNLayerPtr &layer, const InferenceEngine::Blob::Ptr &blob, const unsigned long resized_im_h,
+                       const unsigned long resized_im_w, const unsigned long original_im_h,
+                       const unsigned long original_im_w,
+                       const double threshold, std::vector<DetectionObject> &objects) {
+    // --------------------------- Validating output parameters -------------------------------------
+    if (layer->type != "RegionYolo")
+        throw std::runtime_error("Invalid output type: " + layer->type + ". RegionYolo expected");
+    const int out_blob_h = static_cast<int>(blob->getTensorDesc().getDims()[2]);
+    const int out_blob_w = static_cast<int>(blob->getTensorDesc().getDims()[3]);
+    if (out_blob_h != out_blob_w)
+        throw std::runtime_error("Invalid size of output " + layer->name +
+        " It should be in NCHW layout and H should be equal to W. Current H = " + std::to_string(out_blob_h) +
+        ", current W = " + std::to_string(out_blob_h));
+    // --------------------------- Extracting layer parameters -------------------------------------
+    auto num = layer->GetParamAsInt("num");
+    auto coords = layer->GetParamAsInt("coords");
+    auto classes = layer->GetParamAsInt("classes");
+    std::vector<float> anchors;
+
+    try { anchors = layer->GetParamAsFloats("anchors"); } catch (...) {}
+    try {
+        auto mask = layer->GetParamAsInts("mask");
+        num = mask.size();
+
+        std::vector<float> maskedAnchors(num * 2);
+        for (int i = 0; i < num; ++i) {
+            maskedAnchors[i * 2] = anchors[mask[i] * 2];
+            maskedAnchors[i * 2 + 1] = anchors[mask[i] * 2 + 1];
+        }
+        anchors = maskedAnchors;
+    } catch (...) {}
```

Comment:
I don't think we should be doing all this for every frame, since the results will always be the same. Not only is it unnecessary calculation, but it also prevents us from destroying the `CNNNetReader` object after the network is loaded to a device.

I think something like this would work better:

1. Add a new field to `IEGraph::InitParams` (let's say `postLoadFunc`) whose type is a function that takes a `const CNNNetReader&` argument.
2. In `IEGraph::initNetwork` call this function after the network is loaded.
3. In this demo, initialize `postLoadFunc` with something like `[&yoloParams](const CNNNetReader &reader) { yoloParams = GetYoloParams(reader)`, where `yoloParams` is a local variable in `main`, and `GetYoloParams` is a function with code equivalent to what's above.

Reply:
Thanks, Let me think about how to implement it since the `mask` is different.